test/unit/payment_gateways/brain_tree_blue_crypt_test.rb

```diff
@@ -43,33 +43,21 @@ def test_errors
     end
 
     test '#confirm' do
-      query_string = 'a=1&b=2'
-      request = mock do
-        expects(:query_string).returns(query_string)
-      end
-      @braintree.gateway.expects(:confirm).with(query_string)
-      @braintree.confirm(request)
+      @braintree.expects(:current_token).returns('token')
+      @braintree.customer_gateway.expects(:update).with(@braintree.buyer_reference_for_update, {payment_method_nonce: 'nonce', credit_card: { options: {make_default: true, update_existing_token: 'token'}}})
+      @braintree.confirm({}, 'nonce')
```

Comment:
This method is what is used in the controller to so later `if result && result.success?` so we should probably end up asserting something about the result of this method here as well 😄 
In that case of course it can be later on 👍